pkg/agent/sysadvisor/metacache/metacache.go

```diff
@@ -287,8 +288,18 @@ func (mc *MetaCacheImp) RangeAndUpdateContainer(f func(podUID string, containerN
 	}
 
 	mc.mutex.Lock()
-	mc.podEntries = podEntries
-	mc.mutex.Unlock()
+	defer mc.mutex.Unlock()
```

Comment:
add comments here: during range process, pod entries in metaCache may have already been deleted, so we will only update entries that still exists to avoid race condition